ribbon/src/main/java/feign/ribbon/RibbonClient.java

```diff
@@ -26,6 +28,16 @@
  */
 public class RibbonClient implements Client {
 
+  private static final LoadingCache<String, LBClient> LBCLIENT_CACHE = CacheBuilder.newBuilder()
+      .build(new CacheLoader<String, LBClient>() {
+        @Override
+        public LBClient load(String clientName) throws Exception {
```

Comment:
guava code makes me nervous (eventhough ribbon depends on it). Also static makes me nervous.

if ribbon wants to coordinate everything in the VM, we could propagate that advice by saying "use a single instance of RibbonClient() for all feigns in the VM". This removes us from having to do so explicitly by making a static cache.

Next, we could move load into a synchronized method getOrCreateLBClient that does a guarded check on an internal linkedhashmap cache. if the key is not present, populate it then return the value.

I'd be surprised if we would need to optimize getOrCreateLBClient to be something besides just synchronized.

Make sense?